packages/flutter_driver/lib/src/extension.dart

```diff
@@ -96,6 +99,18 @@ class _FlutterDriverExtension {
   final Map<String, CommandDeserializerCallback> _commandDeserializers = <String, CommandDeserializerCallback>{};
   final Map<String, FinderConstructor> _finders = <String, FinderConstructor>{};
 
+  /// With [_frameSync] enabled, flutter_driver will wait to perform an action
+  /// until there are no pending frames in the app under test.
+  ///
+  /// This greatly reduces flakiness as it prevents flutter_driver from
+  /// interacting with an app that is currently in a transition. However,
+  /// in same cases, an app will never reach a state where there are no pending
+  /// frames (e.g. when there is a continuous animation on the screen). For
+  /// these cases, [_frameSync] needs to be disabled and its the responsibility
+  /// of the test writer to make sure that no flutter_driver actions are
+  /// performed while the app is in a transitional state.
```

Comment:
Let's check with @Hixie re the wording of the dartdocs. I think we generally describe what a method does without going too much into specific use-cases? For example, I'm not sure if it's worth talking about the responsibility of the test writer to make sure that there are no driver actions happening in-flight. Perhaps that's the developer's goal, and therefore that's why they disabled frame syncing.